mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionAvailbilityAwareAcivity.java

```diff
@@ -0,0 +1,148 @@
+import static org.openhab.habdroid.ui.OpenHABPreferencesActivity.NO_URL_INFO_EXCEPTION_MESSAGE;
+
+public abstract class ConnectionAvailbilityAwareAcivity extends AppCompatActivity {
+    private static final String TAG = ConnectionAvailbilityAwareAcivity.class.getSimpleName();
+    private boolean noConnectionFragmentShown = false;
+
+    @Override
+    protected void onStop() {
+        super.onStop();
+
+        try {
+            unregisterReceiver(ConnectionFactory.getInstance());
+        } catch (IllegalArgumentException e) {
+            Log.d(TAG, "Tried to unregister not registered BroadcastReceiver.", e);
+        }
+    }
+
+    public Connection getConnection(int connectionType) {
+        try {
+            return ConnectionFactory.getConnection(connectionType, this);
+        } catch (NetworkNotAvailableException | NetworkNotSupportedException e) {
+            showNoNetworkFragment(e.getMessage());
+        } catch (NoUrlInformationException e) {
+            Intent preferencesIntent = new Intent(this, OpenHABPreferencesActivity.class);
+            preferencesIntent.putExtra(NO_URL_INFO_EXCEPTION_EXTRA, true);
+            preferencesIntent.putExtra(NO_URL_INFO_EXCEPTION_MESSAGE, e.getMessage());
+
+            TaskStackBuilder.create(this)
+                    .addNextIntentWithParentStack(preferencesIntent)
+                    .startActivities();
+        }
+        return null;
+    }
+
+    private void showNoNetworkFragment(String message) {
+        Fragment noNetworkFrament = new NoNetworkFragment();
+        Bundle bundle = new Bundle();
+        bundle.putString(NoNetworkFragment.NO_NETWORK_MESSAGE, message);
+        noNetworkFrament.setArguments(bundle);
+
+        onContentReplace();
+
+        getFragmentManager()
+                .beginTransaction()
+                .replace(android.R.id.content, noNetworkFrament)
+                .commit();
+        noConnectionFragmentShown = true;
+
+        setTitle(R.string.app_name);
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(false);
+        }
+    }
+
+    /**
+     * This method is called whenever this abstract class will replace the whole content with
+     * other information, such as error fragments or something like that. A class extending from
+     * here can override this method to also reset/clear/remove views from the activity if needed.
+     */
+    protected void onContentReplace() {}
+
+    private void restartApp() {
+        Intent startActivity = new Intent(this, OpenHABMainActivity.class);
+        startActivity.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        startActivity(startActivity);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        if (!noConnectionFragmentShown) {
+            return;
+        }
+        try {
+            ConnectionFactory.getConnection(Connection.TYPE_ANY, this);
+            restartApp();
```

Comment:
Why not simply remove the no network fragment?

Reply:
Good idea, done! :)